### ISSUE TITLE
"Trace JSON"/"Trace JSON (unadjusted)" open outside Jaeger UI application

### ISSUE THREAD

**Issue by th3M1ke** [jaegertracing/jaeger-ui#680]
## Requirement - what kind of business use case are you trying to solve?
<img width="355" alt="Screenshot 2021-01-13 at 22 04 21" src="https://user-images.githubusercontent.com/45467774/104920620-b5858d00-59a0-11eb-9636-c0c84425895d.png">

"Trace JSON"/"Trace JSON (unadjusted)" open outside Jaeger UI application. For this case, we have no ability to manage auth tokens in HTTP headers via iframe

## Problem - what in Jaeger blocks you from solving the requirement?

"Trace JSON"/"Trace JSON (unadjusted)" use a direct API call to get JSON from the backend. No ability to handle HTTP headers

## Proposal - what do you suggest to solve the problem or improve the existing situation?

The solution is to create a react component, which makes a call to the backend via JaegerAPI. Also, this component can deprecate "Trace JSON (unadjusted)" in the trace page's dropdown. This functionality can be performed by these components. Also, we can make some progress here https://github.com/jaegertracing/jaeger-ui/issues/197

**Comment by yurishkuro** [comment]
Sorry, I don't fully follow what is being proposed. Are you suggesting a dedicated component to manually render JSON without leaving the UI? Currently View JSON buttons double as a means to download the trace JSON, not only view it.

**Comment by th3M1ke** [comment]
Correct! I suggest a custom component for render/download raw traces. 

The reason for that is once you open "Trace JSON" its go out of UI app control and fire direct API query, which is under control of iframe in our case... 😞

**Comment by yurishkuro** [comment]
I think we may already have some form of JSON rendering for tag values, so perhaps it's not too bad to implement it for full trace JSON. The only concern I have is to make sure that the component scales up for very large traces, since the current method just relies on the browser file rendering with minimal overhead.

**Comment by th3M1ke** [comment]
What is the max size of trace have you face? BTW, you have mentioned `form of JSON rendering for tag values` can you help to find it?

**Comment by yurishkuro** [comment]
* 100,000 spans and more.
* I think this is related PR #162. I.e. when the tag value is a JSON string, we pretty-print it.

**Comment by th3M1ke** [comment]
@yurishkuro `json-markup` almost ok on chrome with hotROD data, but it renders more than 10 seconds in safari... Can you share some examples of trace for 100,000 spans, or just say its approximate size?

**Comment by yurishkuro** [comment]
Sorry I don't have examples, it was with Uber's production data.

Do you mean the size in bytes? For 100,000 spans it would be 50-100Mb.

**Comment by nofar9792** [comment]
Hi @yurishkuro :)
I would like to fix it.
Just want to be sure about the solution: 
1. remove "Trace JSON (unadjusted)" 
2. show the JSON on the same page (and not a new tab) 
![image](https://user-images.githubusercontent.com/11076023/226169081-45e21313-ee78-4af4-a215-9fdd4ca1be29.png)

3. maybe we can add a lib to show the JSON in an interactive way? for example https://www.npmjs.com/package/react-json-view

WDYT?

**Comment by yurishkuro** [comment]
This ticket looks like a duplicate of #1211, which has an in-progress PR solving it.

@nofar9792  As for your specific suggestions:
1. No, we won't remove unadjusted, it's valuable in debugging
2. I don't believe embedding the JSON view is a good solution. It prevents one from downloading the result as JSON, which is currently possible.
